packages/angular_devkit/core/src/jobs/README.md

```diff
@@ -0,0 +1,319 @@
+`createGroup()` function:
+
+```typescript
+import { jobs } from '@angular-devkit/core';
+
+// A group that installs node modules.
+const group = jobs.createGroup({
+  name: 'node-install',
+  input: {
+    properties: {
+      moduleName: { type: 'string' },
+    },
+  },
+  output: { type: 'boolean' },
+});
+
+// Extending makes sure the input and output matches the group.
+const npmInstall = jobs.createJob(/* ... */, { name: 'npm-install', extends: group });
+const yarnInstall = jobs.createJob(/* ... */, { name: 'yarn-install', extends: group });
+const pnpmInstall = jobs.createJob(/* ... */, { name: 'pnpm-install', extends: group });
+
+declare const registry: jobs.SimpleJobRegistry;
+registry.register(group);
+registry.register(npmInstall);
+registry.register(yarnInstall);
+registry.register(pnpmInstall);
+
+// Default to npm.
+group.setDefaultDelegate(npmInstall.name);
+// If the user is asking for yarn over npm, uses it.
+group.addConditionalDelegate(() => userWantsYarn, yarnInstall.name);
+```
+
+## Execution Strategy
+
+Jobs are always run in parallel and will always start, but many helper functions are provided 
+when creating a job to help you control the execution strategy;
+
+1. `serialize()`. Multiple runs of this job will be queued with each others.
+1. `reuse(replayEvents = false)`. Jobs with this strategy will reuse an already running job. If 
+`replayEvents` is true, all events will be replayed in the same order.
+1. `memoize(replayEvents = false)` will create a job, or reuse the same job  when inputs are 
+matching. If the inputs don't match, a new job will be started and its outputs will be stored.
+
+These strategies can be used when creating the job:
+
+```typescript
+// Same input and output as above.
+
+export const add = jobs.strategy.memoize()(
+  jobs.createJob<number[], number>(
+      input => input.reduce((total, curr) => total + curr, 0),
+  ),
+);
+```
+
+Strategies can be reused to synchronize between jobs. For example, given jobs `jobA` and `jobB`, 
+you can reuse the strategy to serialize both jobs together;
+
+```typescript
+const strategy = jobs.strategy.serialize();
+const jobA = strategy(jobs.createJob(...));
+const jobB = strategy(jobs.createJob(...));
+```
+
+Even further, we can have package A and package B run in serialization, and B and C also be 
+serialized. Running A and C will run in parallel, while running B will wait for both A and C 
+to finish.
+
+```typescript
+const strategy1 = jobs.strategy.serialize();
+const strategy2 = jobs.strategy.serialize();
+const jobA = strategy1(jobs.createJob(...));
+const jobB = strategy1(strategy2(jobs.createJob(...)));
+const jobC = strategy2(jobs.createJob(...));
+```
+
+# Scheduling Jobs
+Jobs can be scheduled using a `Scheduler` interface, which contains a `schedule()` method. The
+method takes a name, an input and a list of options.
+
+## Synchronizing and Dependencies
```

Comment:
Isn't synchronization managed by the execution strategy? 

Are these different things, and if so, what are the different use cases?

Reply:
These are different things. The difference is that strategy is set by the actual Job code, while this synchronization is done by the person scheudling the job. Clarfiied.